os/mm/umm_heap/umm_calloc.c

```diff
@@ -121,6 +126,21 @@ FAR void *calloc(size_t n, size_t elem_size)
 			return ret;
 		}
 	}
+
+#if (defined(CONFIG_RAM_MALLOC_PRIOR_INDEX) && CONFIG_RAM_MALLOC_PRIOR_INDEX > 0)
+	/* Try to mm_calloc to other heaps */
+	for (heap_idx = 0; heap_idx < CONFIG_RAM_MALLOC_PRIOR_INDEX; heap_idx++) {
```

Comment:
I think it is not good to place almost same code.
How about making static function for traversing the heap region?

Reply:
updated static function for each